docs/DATASETS_v3.md

```diff
@@ -0,0 +1,355 @@
+
+#### Geometry encoding
+
+Geometries are converted to byte strings before they are serialised using MessagePack. The geometry bytestring is marked as being a MessagePack extension with the extension code `"G"` (71). The encoding used to serialise the geometry is as follows.
+
+Geometries are encoded using the Standard GeoPackageBinary format specified in [GeoPackage v1.3.0 §2.1.3 Geometry Encoding](http://www.geopackage.org/spec/#gpb_data_blob_format), with additional restrictions:
+
+1. Geometries must use the StandardGeoPackageBinary type.
+2. GeoPackage binary headers must always use little-endian byte ordering.
+3. The WKB geometry must always use little-endian byte ordering.
+4. All non-empty geometries must have an envelope, except for POINT types:
+   - Points and empty geometries have no envelope.
+   - Geometries with a Z component have an XYZ envelope.
+   - Other geometries have an XY envelope.
+5. The `srs_id` is always 0, since this information not stored in the geometry object but is stored on a per-column basis in `meta/schema.json` in the `geometryCRS` field.
+
+**Note on axis-ordering:** As required by the GeoPackageBinary format, which Kart uses internally for geometry storage, Kart's axis-ordering is always `(longitude, latitude)`. This same axis-ordering is also used in Kart's JSON and GeoJSON output.
+
+### Feature paths
+
+Every feature is stored at a path based on its primary key, so that an update to the feature that doesn't change its primary key will cause it to be overwritten in place. The primary key value can be transformed into its path and back into a primary key value without losing any information - for this reason, the values for primary key columns are not included in the contents of a feature file, since they can be inferred from the file's name.
+
+A feature path might look like this:
+
+`A/A/A/B/kU0=`
+
+There are two parts to this: the path to the file - `A/A/A/B` - and the filename itself - `kU0=`. 
+
+#### Feature path filename
+
+The filename is the more important part, and it is generated in the following manner:
+
+`urlsafe_b64encode(msgpack.packb(primary_key_value_array))`
+
+In the example feature path above, there is only one primary key column, and the feature being stored is the feature with primary key 77. So the primary key values are an array of length one containing 77: `[77]`. So the filename was generated as follows:
+
+`[77]` -> MessagePack -> `bytes([0x91, 0x4d])` -> Base64 -> `kU0=`
+
+#### Path to the feature file
+
+For technical reasons, it is best if only a relatively small number of features are stored together in a single directory, and similarly if only a small number of directories are stored together in a single directory. Ideally, the features created at the same time or likely to be edited at the same time should be stored together, rather than spread out among all the other features - so, neighbouring primary key values should be neighbouring files where possible.
```

Comment:
```suggestion
For technical reasons, it is best if only a relatively small number of features are stored together in a single directory, and similarly if only a small number of directories are stored together in a single directory. Ideally, the features created at the same time or likely to be edited at the same time should be stored together, rather than spread out among all the other features - so, neighbouring primary key values should be neighbouring file paths where possible.
```

Reply:
Done